### PR TITLE
[FEAT] Add new fail_on_warnings to the coding-standard action

### PR DESCRIPTION
**Pull request by vitaliy-golomoziy**
A PR for the issue https://github.com/mage-os/github-actions/issues/130

Fixed: #130

### CODE REVIEW

**Review by damienwebdev — CHANGES_REQUESTED** [review]
```
--config-set ignore_warnings_on_exit 1
```

Is better than changing the warning severity I think. 

This way, it will still echo out the warnings, but we won't get the non-zero exit status.

**Comment by damienwebdev** [comment]
We'll need to allow the plugin by the demo package: https://github.com/mage-os/github-actions/blob/main/_test/demo-package/composer.json#L44

File: _test/demo-package/composer.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
         "preferred-install": "dist",
         "sort-packages": true,
         "allow-plugins": {
-            "magento/composer-dependency-version-audit-plugin": true
+            "magento/composer-dependency-version-audit-plugin": true,
+	    "dealerdirect/phpcodesniffer-composer-installer": true
```
Comment (damienwebdev) [inline _test/demo-package/composer.json:45]: This can be removed.

File: coding-standard/action.yml [hunk 2]
```diff
@@ -37,6 +37,11 @@ inputs:
     default: ""
     description: "The minimum severity required to display an error"
 
+  fail_on_warnings:
+    description: 'Whether action should fail on warnings or not, default to true (fails)'
```
Comment (damienwebdev) [inline coding-standard/action.yml:41]: Grammar.

> Whether or not the action should fail on warnings, defaults to true (fails on warnings)

**Review by damienwebdev — CHANGES_REQUESTED** [review]

**Review by damienwebdev — APPROVED** [review]
lgtm

File: coding-standard/action.yml [hunk 3]
```diff
@@ -37,6 +37,11 @@ inputs:
     default: ""
     description: "The minimum severity required to display an error"
 
+  fail_on_warnings:
+    description: 'Whether or not the action should fail on warnings, defaults to true (fails on warnings)'
+    default: 1
```
Comment (damienwebdev) [inline coding-standard/action.yml:42]: Could this be `true` instead of `1`. While I am a robot, not everyone else is!

**Review by damienwebdev — CHANGES_REQUESTED** [review]

**Comment by vitaliy-golomoziy** [comment]
@damienwebdev, please review

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: semver-compare/action.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    description: "Original version"
+
+  compare_against:
+    required: true
+    description: "The version to compare against"
+
+outputs:
+  result: # id of output
+    description: 'The result of comparison'
+    value: ${{ steps.semver-compare.outputs.result }}
```
[the remark this answers is not in the document]
Comment (damienwebdev) [reply semver-compare/action.yml:17]: @vitaliy-golomoziy bump on this.

**Comment by damienwebdev** [comment]
Lookl like this PR has to be split into smaller PRs because of dependencies, the `@main` usage of a currently non-existing action inside of `coding-standard.yml` is failing the pipeline.

**Review by damienwebdev — CHANGES_REQUESTED** [review]
Looks like this PR has to be split into smaller PRs because of dependencies, the `@main` usage of a currently non-existing action inside of `coding-standard.yml` is failing the pipeline.

`get-composer-version` and `semver-compare` look good to me, so these can be split and merged quickly.

File: semver-compare/action.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (vitaliy-golomoziy) [reply semver-compare/action.yml:17]: Correct, and also missed the README.md update, fixed both now.

**Comment by damienwebdev** [comment]
Closing in favor of #145 #146 and #147

**Pull request closed without merging**